docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
+        c. From here on, you can think of Runhouse as facilitating
+            regular object-oriented programming but with the objects living remotely, maybe in a different cluster,
+            region, or cloud than the local code.
+        d. Python behavior like async, exceptions, printing, and logging are all preserved across remote calls, but
+            can be disabled or controlled if desired.
+5. **Saving and Loading**: The Runhouse client can save and load objects to and from the local filesystem, or to a
+    remote metadata store. This allows for easy sharing of clusters and services across users and environments,
+    and for versioning and rollback of objects. The metadata store can be accessed from any Python interpreter,
+    and is backed by a management API to view and manage all resources.
+6. **Terminating Modules, Workers, or Clusters**: When a remote object is no longer needed, it can be deallocated from
+    the remote compute by calling `cluster.delete(obj_name)`. This will remove the object from the key-value store and
+    free up the memory on the worker. A full worker can similarly be terminated with `cluster.delete(worker_name)`. A
+    cluster can be terminated with `cluster.down()`.
+
+Comparing to other systems
+~~~~~~~~~~~~~~~~~~
 
-* `Runhouse Den <https://www.run.house/dashboard>`_: provides an individual or admin view of all resources, secrets,
-  groups, and sharing. Resource metadata is automatically versioned in RNS, allowing teams to maintain single-sources
-  of truth for assets with zero downtime to update or roll back, and trace exact lineage for any existing resource.
+Runhouse's APIs bear similarity to other systems, so it's helpful to compare and contrast. In many cases,
+Runhouse is not a replacement for these systems, but rather a complement or extension. In others, you may be able
+to replace your usage of the other system entirely with Runhouse.
+
+*Distributed frameworks* like Ray, Spark, or Elixr make it possible to offload execution onto separate
+compute, like a different process or node within a their cluster runtime. Runhouse
+can be seen as similar, but with the crucial distinction of dispatching execution to compute *outside* of its own
+runtime (which is just Python) or orchestrating *between* clusters (even of different types).
+For this reason, it has no other runtime to setup than Python itself, can be used to orchestrate your distributed code so you
+can use your Ray or Spark clusters less disruptively within your stack (e.g. sending a function which uses
+Ray over to the head node of the Ray cluster, where the Ray will execute as usual).
+
+<diagram showing Ray calling actors within the cluster, Runhouse calling outside into other processes, VMs,
+Ray clusters, K8s clusters, clouds>
+
+This also fixes some other sharp
+edges with such systems which significantly reduces costs, such as utilizing more than one cluster in an application,
```

Comment:
This also fixes sharp edges and inefficiencies with these systems, such as the inability to use more than one cluster in an application or sharing a cluster between multiple callers.